### ISSUE TITLE
Failure during import of large (~3GB) S3 files

### ISSUE THREAD

**Issue by joelarmstrong** [BD2KGenomics/toil#1672]
It looks to me like the s3 copier may need to block until the upload ID is created. Sometimes the first few parts will fail copying, but the rest will succeed:

```
Initiated multipart copy from 's3://cactus-alignments/tenway--Chimp' to 's3://cactus-jcarmstr-tenway-pp--files/86d1d2f8-6eb2-4dfb-a6eb-171f0b9b8a1a'.
Failed to copy part number 12:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 96, in copyPart
    headers=headers)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/multipart.py", line 306, in copy_part_from_key
    query_args=query_args)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/bucket.py", line 888, in copy_key
    response.reason, body)
S3ResponseError: S3ResponseError: 404 Not Found
<Error><Code>NoSuchUpload</Code><Message>The specified upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.</Message><UploadId>fVDG2y81z0Ptix9dK0cwu.K228.m8gbA8$
faZOAhWMrRbxd_xkvBRpqJu3kzSb8Dqg4j9i3v3H9VOGL36wY6BSVCUH6I8cOZFop9jV52weUk5Jp2Def_9pTfxJMRtdKZfTF1J.Iuztw_Yz.vYrXSR2.V_pyTapP6MU.ESiRFNyA-</UploadId><RequestId>116C4E2F81B7F0B4</RequestId><HostId>B6cGk8MZb7uowe$
/Vr4mjvEORiQ42lebHtaRFHrq5SxwqNddoMDoY15W98oibpVJ8nrOOOvM62U=</HostId></Error>
Failed to copy part number 10:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 96, in copyPart
    headers=headers)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/multipart.py", line 306, in copy_part_from_key
    query_args=query_args)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/bucket.py", line 888, in copy_key
    response.reason, body)
S3ResponseError: S3ResponseError: 404 Not Found
<Error><Code>NoSuchUpload</Code><Message>The specified upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.</Message><UploadId>fVDG2y81z0Ptix9dK0cwu.K228.m8gbA8$
faZOAhWMrRbxd_xkvBRpqJu3kzSb8Dqg4j9i3v3H9VOGL36wY6BSVCUH6I8cOZFop9jV52weUk5Jp2Def_9pTfxJMRtdKZfTF1J.Iuztw_Yz.vYrXSR2.V_pyTapP6MU.ESiRFNyA-</UploadId><RequestId>9F40C320BE4F703C</RequestId><HostId>yb7FrZMlM7NvBB$
jTqwomrxUbo8z3cavVuCaRX8X9cLS9L9c9G54p3B/o+hwOvBI0jQWoRgpFps=</HostId></Error>
Failed to copy part number 18:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 96, in copyPart
    headers=headers)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/multipart.py", line 306, in copy_part_from_key
    query_args=query_args)
  File "/usr/local/lib/python2.7/dist-packages/boto/s3/bucket.py", line 888, in copy_key
    response.reason, body)
S3ResponseError: S3ResponseError: 404 Not Found
<Error><Code>NoSuchUpload</Code><Message>The specified upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.</Message><UploadId>fVDG2y81z0Ptix9dK0cwu.K228.m8gbA8$
faZOAhWMrRbxd_xkvBRpqJu3kzSb8Dqg4j9i3v3H9VOGL36wY6BSVCUH6I8cOZFop9jV52weUk5Jp2Def_9pTfxJMRtdKZfTF1J.Iuztw_Yz.vYrXSR2.V_pyTapP6MU.ESiRFNyA-</UploadId><RequestId>DDAE108168AD081A</RequestId><HostId>lvWhbj8EC8yBjZ$
+s2pRdy9dMN7VpnKIBEF+MSnLoxJxLNwAN33a0CF8rkQE0C7d685m/ZJy4XU=</HostId></Error>
Successfully copied part 8 of 63.
Successfully copied part 14 of 63.
Successfully copied part 4 of 63.
Successfully copied part 13 of 63.
Successfully copied part 3 of 63.
```

**Comment by cket** [comment]
Interesting.. Is this on a relatively up-to-date Toil version? Previously we weren't blocking on uploads but I believe I fixed that issue. I'll see if I can dig up the issue/PR to see if they're related to this.

**Comment by joelarmstrong** [comment]
Hi CJ,

Sorry for the long delay. My toil version has everything in master up to ~ April 1st. I had looked through the unmerged commits and didn't see anything that seemed to be relevant at first glance.

Regardless, I wasn't able to trigger this bug on the latest toil master branch. So it's gotta be something in the unmerged commits, or some conflict that I resolved improperly. Either way, it's on my end, so I'm closing this.
